code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -225,13 +256,35 @@
 
 /obj/structure/closet/examine(mob/user)
 	. = ..()
-
-	. += span_notice("It can be [EXAMINE_HINT("welded")] apart.")
-	. += span_notice("It can be [EXAMINE_HINT("bolted")] to the ground.")
-
+	. += "You can change its name & description with a pen."
+	if(welded)
+		. += span_notice("It's welded shut.")
+	else if(can_weld_shut)
+		. += span_notice("It can be welded shut.")
+	if(anchored)
+		. += span_notice("It is <b>bolted</b> to the ground.")
+	if(opened && cutting_tool == /obj/item/weldingtool)
+		. += span_notice("The parts are <b>welded</b> together.")
+	if(secure && !opened)
+		. += span_notice("Right-click to [locked ? "unlock" : "lock"].")
+	if(length(paint_jobs))
+		. += span_notice("Use an airlock painter to change its texture.")
 	if(HAS_TRAIT(user, TRAIT_SKITTISH) && divable)
 		. += span_notice("If you bump into [p_them()] while running, you will jump inside.")
 
+	if(can_install_electronics)
+		if(!secure)
+			. += span_notice("You can install airlock electronics for access control.")
+		else
+			. += span_notice("Its airlock electronics are [EXAMINE_HINT("screwed")] in place.")
+
+		if(!card_reader_installed && length(access_choices))
+			. += span_notice("You can install a card reader for furthur access control.")
+		else if(card_reader_installed)
+			. += span_notice("The card reader could be [EXAMINE_HINT("pried")] out.")
+			. += span_notice("Swipe your PDA with an ID card/Just ID to change access levels.")
+			. += span_notice("Use multitool to [access_locked ? "unlock" : "lock"] the access panel.")
+
```

Comment:
Even with my suggestion, I still feel like it's very verbose.  Might want to consider getting rid of the card reader hints.

Reply:
But how will people know what the card reader does? 

```
if(!card_reader_installed && length(access_choices))
			. += span_notice("You can install a card reader for furthur access 
```
the length() part is nessassary because some genpop closets will set this to null because we don't want to change their functioning with a card reader, I'll copy the hints from here that makes sense.

`if(can_install_electronics)`
We need this too, because this variable is set to false by graves[which are a subset of closets weirdly] we can tell the player its possible to install electronics in graves
https://github.com/tgstation/tgstation/blob/dbd6f71d6ab18acc4c5711e2866464b1d472bc4c/code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm#L144

I'm sorry but like you said examines need to show only what's possible and a lot of your suggested examines don't check for these variables, this doesn't seem salvageable

Reply:
Well will try to salvage what we can

Reply:
Ok let me give an example:

`. += span_notice("It can be [EXAMINE_HINT("bolted")] to the ground.")`

Using a wrench to bolt a locker to the ground is not possible for all lockers nor do we care if the locker is currently bolted or not.  See:

https://github.com/tgstation/tgstation/blob/046a5cbe01e5c8a06337b7fbdc376e4765e94b18/code/game/objects/structures/crates_lockers/closets.dm#L533-L543

Your card reader hints should do the same thing.  We should try and let balloon alerts do the talking and combine it with contextual screentips.  You may even consider adding the information to the card reader description itself.

The hint for the card reader already conveys a decent amount of information.

`. += span_notice("It can be [EXAMINE_HINT("installed")] with airlock electronics or a card reader to control access.")`

Reply:
I see, well since I already made the procs for these checks to use in do_after() it should be fairly easy to edit, I'll do that

Reply:
Ok so after taking your points into consideration i have decided that it makes sense in some places to tell the player that something is possible but then allow the balloon alert to tell you "The Next Step" in making it possible and not "Its impossible altogether" .

Like take the wrenching a closet for example,
`. += span_notice("It can be [EXAMINE_HINT("bolted")] to the ground.")`

Telling a player that a grave can be wrenched to the ground is just weird? even though the balloon alert won't let you to do that it's still dumb to tell the player they can wrench a grave in the examine, Same thing goes for welding a grave to the ground,
`. += span_notice("It can be [EXAMINE_HINT("welded")] apart.")`
 like we can't tell them it "could" be possible to weld a grave(which already sounds like nonsense) only to later realize it's not like we built up their hopes for nothing. 

My idea is that if a "meaningful action" is possible "for that given type" it should be shown in the examines & the balloon alert should only tell you what's the "next step" to perform that action and should not say "its impossible forever"

Same thing for electronics & card reader, only when they are not installed should we tell them it's possible to install it but when it's already installed telling them this
`. += span_notice("It can be [EXAMINE_HINT("installed")] with airlock electronics or a card reader to control access.")`
Would make them think it's not installed and they would try to install it again only to realize it's already installed & they would have just wasted their time

> I didn't add extra checks for that bolted hint to keep it short and sweet

While from a coding standpoint this is nice for us, it's not so much for the player. If they examine from a distance & see the closet is bolted then they can save time by getting a wrench before approaching the closet, But if we don't tell them it's bolted then when they try to drag it, they realize it's stuck and then they have to go back to get a wrench and we have made them apply twice the effort(1 effort in dragging the closet only to realize its wrenched, 1 more effort in going back to get a wrench). Verbosity from a player standpoint is what helps people learn(Like me best example).

I hope we can come both agree on these points, I understand it's a very subjective matter but i think a majority of people would agree.